testcafe/tests/helpers/utils.js

```diff
@@ -1,7 +1,49 @@
-import { ClientFunction } from 'testcafe'
+import { ClientFunction, t } from 'testcafe'
 
 export const TESTCAFE_PHOTOS_URL = process.env.TESTCAFE_PHOTOS_URL
 export const TESTCAFE_USER_PASSWORD = process.env.TESTCAFE_USER_PASSWORD
 
 //Returns the URL of the current web page
 export const getPageUrl = ClientFunction(() => window.location.href)
+
+export function wait(timeout) {
+  //@param {number} timeout : amount of time in milliseconds during which a selector property must be stable
+  return {
+    until: selector => {
+      return {
+        // exists property of selector
+        exists: {
+          // wait until exists property of selector is stable
+          isStable: async () => {
+            const interval = 100
+            const elapsedTimeMaxValue = timeout * 3
+            let stabilityElapsedTime = 0
+            let previousValue = ''
+            let elapsedTime = 0
+            while (true) {
+              await t.wait(interval)
+              const currentValue = (await selector.exists).toString()
+              //console.log(`Selector exists property evaluates to '${currentValue}'`)
+              if (currentValue !== previousValue) {
```

Comment:
For my own reference, why does return value of a selector change over time? Naively I would think that once the element is available in the DOM, the selector will always return the same thing?